### PR TITLE
Add \mathit where needed

### PR DESCRIPTION
**Pull request by andreas-junghanns**
I added \mathit where needed, I think.

### CODE REVIEW

**Comment by andreas-junghanns** [comment]
This addresses #915. I fail to link this issue in the side bar... no idea what is wrong.

File: docs/2_1_common_api.adoc [hunk 1]
```diff
@@ -500,9 +500,9 @@ All elements of the array, including structural zeros, are serialized.
 _[Example: A 2D matrix_
 [latexmath]
 ++++
-A = \left( \begin{array}{cc} a_{11}&a_{12}\\
-                             a_{21}&a_{22}\\
-                             a_{31}&a_{32}\\
+A = \left( \begin{array}{cc} a_{\mathit{11}}&a_{\mathit{12}}\\
```
Comment (clagms) [inline docs/2_1_common_api.adoc:503]: I think there's no need to use \mathit on numbers. It's only on text, as that's where latex does funny stuff.
This comment applies to everywhere else where changes were made.

File: docs/2_1_common_api.adoc [hunk 2]
```diff
@@ -587,19 +587,19 @@ A clocked partition is mathematically defined as:
...
 [latexmath]
 ++++
-y_j=g_j(x_{j-1},u_j,t_j);
+y_j=g_j(x_{\mathit{j-1}},u_j,t_j);
 ++++
 
 [latexmath]
 ++++
-\text{with } j=0,1,2,...;x_{-1}=x_{start}
+\text{with } j=0,1,2,...;x_{\mathit{-1}}=x_{\mathit{start}}
```
Comment (clagms) [inline docs/2_1_common_api.adoc:602]: Here, only \mathit{start} is correct.

File: docs/2_1_common_api.adoc [hunk 3]
```diff
@@ -960,12 +960,12 @@ In every state, the general form of the FMU equations are:
...
 [latexmath]
 ++++
-\mathbf{v}_{unknown} = \mathbf{h}(\mathbf{v}_{known}, \mathbf{v}_{rest}),
+\mathbf{v}_{\mathit{unknown}} = \mathbf{h}(\mathbf{v}_{\mathit{known}}, \mathbf{v}_{\mathit{rest}}),
 ++++
 
 where
 
-* latexmath:[\mathbf{v}_{unknonwn}] is the vector of unknown floating point variables computed in the actual state:
+* latexmath:[\mathbf{v}_{\mathit{unknonwn}}] is the vector of unknown floating point variables computed in the actual state:
```
Comment (clagms) [inline docs/2_1_common_api.adoc:968]: Typo here: is it on purpose that we're keeping that typo?

File: docs/2_2_common_schema.adoc [hunk 4]
```diff
@@ -1912,22 +1912,22 @@ In other words, the respective partial derivatives cannot be computed.
 |
 [[dependenciesKind, `dependenciesKind`]]
 If <<dependenciesKind>> is present, <<dependencies>> must be present and must have the same number of list elements.
-If not present, it must be assumed that the unknown latexmath:[\color{blue}{v_{\text{unknown}}}] depends on the knowns latexmath:[\color{blue}{\mathbf{v}_{\text{known}}}] without a particular structure.
-Otherwise, the corresponding known latexmath:[\color{blue}{v_{known,i}}] enters the equation as:
+If not present, it must be assumed that the unknown latexmath:[\color{blue}{v_{\mathit{unknow}}}}] depends on the knowns latexmath:[\color{blue}{\mathbf{v}_{\mathit{known}}}] without a particular structure.
+Otherwise, the corresponding known latexmath:[\color{blue}{v_{\mathit{known,i}}}] enters the equation as:
 
-`= dependent`: no particular structure, latexmath:[\color{blue}{{h(..,\ v}_{known,i}}],..)
+`= dependent`: no particular structure, latexmath:[\color{blue}{{h(..,\ v}_{\mathit{known,i}}}],..)
```
Comment (clagms) [inline docs/2_2_common_schema.adoc:1918]: Maybe this was find and replace magic, but 
`_{\mathit{known,i}}`
can be made more correct if transformed into:
`_{\mathit{known}, i}`
as `i` is a different element.

File: docs/2_4_common_co-simulation.adoc [hunk 5]
```diff
@@ -5,7 +5,7 @@
...
 During time integration, the simulation is again performed independently for all subsystems restricting the data exchange between subsystems to discrete communication points latexmath:[tc_i].
 For simulator coupling, also the visualization and post-processing of simulation data is done individually for each subsystem in its own native simulation tool.
-In different contexts, the communication points latexmath:[tc_i], the communication steps latexmath:[tc_i \rightarrow tc_{i+1}] and the communication step sizes latexmath:[hc_i := tc_{i+1} - tc_i] are also known as sampling points (synchronization points), macro steps and sampling rates, respectively.
+In different contexts, the communication points latexmath:[tc_i], the communication steps latexmath:[tc_i \rightarrow tc_{\mathit{i+1}}] and the communication step sizes latexmath:[hc_i := tc_{\mathit{i+1}} - tc_i] are also known as sampling points (synchronization points), macro steps and sampling rates, respectively.
```
Comment (clagms) [inline docs/2_4_common_co-simulation.adoc:8]: `\mathit{i+1}` is incorrect. Should be only `i+1`. 
Mathit should be used one for words that are meant to denote the same mathematical concept.

**Comment by andreas-junghanns** [comment]
@clagms : why do you list these issues here and not fix it right in the PR?
Listing here makes this extremely inefficient and your comments are likely all valid, so they should not be treated as proposals but as improvements.

File: docs/2_1_common_api.adoc (hunk 1 above)
Comment (andreas-junghanns) [reply docs/2_1_common_api.adoc:503]: fixed

File: docs/2_1_common_api.adoc (hunk 2 above)
Comment (andreas-junghanns) [reply docs/2_1_common_api.adoc:602]: fixed

File: docs/2_1_common_api.adoc (hunk 3 above)
Comment (andreas-junghanns) [reply docs/2_1_common_api.adoc:968]: fixed

File: docs/2_2_common_schema.adoc (hunk 4 above)
Comment (andreas-junghanns) [reply docs/2_2_common_schema.adoc]: fixed

File: docs/2_4_common_co-simulation.adoc (hunk 5 above)
Comment (andreas-junghanns) [reply docs/2_4_common_co-simulation.adoc]: fixed

**Comment by andreas-junghanns** [comment]
@clagms : I addressed all your comments. Can you please check again (and approve)?
I would like to merge soon as this touches many lines and merge conflicts with future PRs are likely.

**Comment by clagms** [comment]
This is not fixed yet @andreas-junghanns , but I'll do it!

**Comment by clagms** [comment]
In the equations near ``An example of use for an array of stream variables is a gas mixture ``, the `\textit` command should be replaced by `\mathit`. However, because I don't know what the `[]` subscript means, I'll leave those equations as they are.
Here's a picture:
image

**Comment by clagms** [comment]
I fixed the mathit's that were introduced incorrectly as part of the regex magic.
I did not check whether there are math elements that should have a mathit in them (I assume @andreas-junghanns did not miss any of these).

**Pull request merged**